client/my-sites/customer-home/locations/primary/index.jsx

```diff
@@ -67,12 +67,20 @@ const cardComponents = {
 	[ TASK_VERIFY_EMAIL ]: VerifyEmail,
 };
 
-const Primary = ( { cards, trackCards } ) => {
+const Primary = ( { cards, trackCard } ) => {
+	const [ currentPage, setCurrentPage ] = useState( 0 );
+	const [ viewedCards, setViewedCards ] = useState( {} );
```

Comment:
Using a `Set` could tidy up some of the logic that deals with ensuring the event only fires once.

I think it's also possible to do this without adding any extra "state" management to this component? React state will queue re-renders but tracking doesn't require any renders (renders would be queued but they'd all be batched so I can't imagine it'd have any performance implications).
I think having a `currentPage` state would also confuse things a little, like my comments above, because it makes it look like the `<DotPager>` is a controlled component, that the parent is the component dealing with the current page state, when it's not.

Idea:

```typescript
const viewedCards = useRef( new Set() );

const handlePageSelected = ( index ) => {
  const selectedCard = cards && cards[ index ];
  if ( viewedCards.current.has( selectedCard ) ) {
    return;
  }

  viewedCards.current.add( selectedCard );
  trackCard( selectedCard );
};

// I guess something like this would deal with tracking the first cards render?
useEffect( () => handlePageSelected( 0 ) );
```